cli/azd/pkg/templates/template.go

```diff
@@ -14,8 +14,12 @@ type Template struct {
 	// Name is the friendly short name of the template.
 	Name string `json:"name"`
 
+	Title string `json:"title,omitempty"`
```

Comment:
In your mind, what is the difference between Title and Name? I think this is a new field we're adding?

Reply:
The field `name` was renamed to `Title` on awesome-azd template list.
I added this to support folks using awesome-azd as an example to create their own template source file. For example, if someone forks the awesome-azd gallery and wants to add it as a source to azd.

Reply:
I also would avoid adding these optional fields.  This `Template` struct is essentially our schema and I don't see any strong need for optionally equivalent properites.

Reply:
What do you think about the use case I shared? 

> someone forks the awesome-azd gallery with their own template list and wants to add it as a source to azd.

without the mapping, azd will not display the template list :(